test/RawRabbit.IntegrationTests/SimpleUse/PublishAndSubscribeTests.cs

```diff
@@ -15,399 +15,509 @@
+                    {
+                        tcs.TrySetResult(message);
+                        return Task.FromResult(true);
+                    }, cfg => cfg
+                    .WithQueue(q => q
+                        .AssumeInitialized()
+                        .WithName(queueName))
+                    .WithExchange(e => e
+                        .AssumeInitialized()
+                        .WithName(exchangeName)));
+
+                    /* Test */
+                    await client.PublishAsync(new DynamicMessage { Body = new { IsDynamic = true } },
+                            configuration: cfg => cfg
+                                .WithExchange(
+                                    e => e
+                                        .AssumeInitialized()
+                                        .WithName(exchangeName)));
+                    await tcs.Task;
+
+                    /* Assert */
+                    Assert.True(tcs.Task.Result.Body.IsDynamic);
+                }
+                finally
+                {
+                    /* Teardown */
+                    TestChannel.QueueDelete(queueName);
+                    TestChannel.ExchangeDelete(exchangeName);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task Should_Throw_When_Subscribed_Not_Initialized_Queue()
+        {
+            using (var client = TestClientFactory.CreateNormal())
+            {
+
+                /* Setup */
+                const string exchangeName = "initialized.exchange";
+                const string queueName = "not.initialized.queue";
+                var tcs = new TaskCompletionSource<DynamicMessage>();
+                TestChannel.ExchangeDeclare(exchangeName, RabbitMQ.Client.ExchangeType.Fanout);
+
+                /* Test */
+                try
+                {
+                    client.SubscribeAsync<DynamicMessage>((message, context) =>
+                    {
+                        tcs.TrySetResult(message);
+                        return Task.FromResult(true);
+                    }, cfg => cfg
+                        .WithQueue(q => q
+                            .AssumeInitialized()
+                            .WithName(queueName))
+                        .WithExchange(e => e
+                            .WithName(exchangeName)));
+
+                    await client.PublishAsync(new DynamicMessage {Body = new {IsDynamic = true}},
+                        configuration: cfg => cfg
+                            .WithExchange(
+                                e => e
+                                    .AssumeInitialized()
+                                    .WithName(exchangeName)));
+                    await tcs.Task;
+
+                    /* Assert */
+                    Assert.True(tcs.Task.Result.Body.IsDynamic);
+                }
+                catch (AggregateException)
+                {
+                    Assert.False(tcs.Task.IsCompleted);
```

Comment:
You could use the `Assert.ThrowsAsync<TException>()` here. Might make the test easier to read?